src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIProjectUserGroupsService.java

```diff
@@ -102,39 +110,88 @@ public String addUserGroupToProject(Long projectId, NewMemberRequest request, Lo
+		/*
+		 If a usergroup's project role is set to collaborator we drop the metadata restriction to the lowest
+		 level and have the project owner set it accordingly. If a role of owner is set then that usergroup is
+		 given full metadata permissions
+		 */
+		if (projectRole.equals(ProjectRole.PROJECT_OWNER)) {
+			projectMetadataRole = ProjectMetadataRole.fromString("LEVEL_4");
+		} else {
+			projectMetadataRole = ProjectMetadataRole.fromString("LEVEL_1");
+		}
 
 		try {
 			projectService.updateUserGroupProjectRole(project, group, projectRole, projectMetadataRole);
-			return messageSource.getMessage("server.usergroups.update.success",
+			return messageSource.getMessage("server.update.projectRole.success",
 					new Object[] { group.getLabel(), roleString }, locale);
 		} catch (ProjectWithoutOwnerException e) {
-			throw new ProjectWithoutOwnerException(
-					messageSource.getMessage("server.usergroups.update-role.error", new Object[] { group.getLabel() },
-							locale));
+			throw new UIProjectWithoutOwnerException(messageSource.getMessage("server.ProjectRoleSelect.error",
+					new Object[] { group.getLabel(), roleString }, locale));
+		} catch (TransactionSystemException e) {
+			ConstraintViolationException constraintViolationException;
+			String constraintViolationMessage = "";
+			if (e.getRootCause() instanceof ConstraintViolationException) {
+				constraintViolationException = (ConstraintViolationException) e.getRootCause();
+				Set<ConstraintViolation<?>> constraintViolationSet = constraintViolationException.getConstraintViolations();
+				for (ConstraintViolation constraintViolation : constraintViolationSet) {
+					// The constraintViolation.getMessage() returns the i18n key in this case
+					constraintViolationMessage +=
+							messageSource.getMessage(constraintViolation.getMessage(), new Object[] {}, locale) + "\n";
+				}
+			}
+			throw new UIConstraintViolationException(constraintViolationMessage);
 		}
 	}
+
+	/**
+	 * Update the {@link ProjectMetadataRole} of a {@link UserGroup} on the current {@link Project}
+	 *
+	 * @param projectId    Identifier for a {@link Project}
+	 * @param groupId      Identifier for an {@link UserGroup}
+	 * @param metadataRole metadata role to update for the group
+	 * @param locale       Current users {@link Locale}
+	 * @return message to user about the result of the update
+	 */
+	public String updateUserGroupMetadataRoleOnProject(Long projectId, Long groupId, String metadataRole,
+			Locale locale) {
+		Project project = projectService.read(projectId);
+		UserGroup group = userGroupService.read(groupId);
+		ProjectMetadataRole projectMetadataRole = ProjectMetadataRole.fromString(metadataRole);
+		String roleString = messageSource.getMessage("metadataRole." + metadataRole, new Object[] {}, locale);
+
+		projectService.updateUserGroupProjectMetadataRole(project, group, projectMetadataRole);
+		return messageSource.getMessage("server.update.metadataRole.success",
```

Comment:
Should this also catch `UIConstraintViolationException`, otherwise I think this would just raise server error if you manually posted this an update when ProjectRole is set to PROJECT_OWNER.